<launch>

    <!-- Spawn Task-4 Models in Gazebo -->
    <include file="$(find pkg_vb_sim)/launch/task4_simulation.launch" />

    <!-- Launch Rviz nodes for two ur5 gazebo arms -->
    <include file="$(find pkg_task4)/launch/two_ur5_move_group.launch" />

    <!-- Group ur5_1 to launch the rviz and their respective node -->
    <group ns="ur5_1">
        <!-- Add Scene file to Moveit! Planning Scene in RViz -->
        <arg name="scene_file" default="$(find pkg_task4)/config/rviz/ur5_1.scene"/>
        <node name = "moveit_publish_scene_from_text" pkg= "moveit_ros_planning" type = "moveit_publish_scene_from_text" args= "$(arg scene_file)"/>
        <!-- Run Task-4 ROS Node for ur5_1 -->
        <node name= "node_t4_1" pkg= "pkg_task4" type="node_t4_1.py" output="screen"/>
    </group>

    <!-- Group ur5_1 to launch the rviz and their respective node -->
    <group ns="ur5_2">
        <!-- Add Scene file to Moveit! Planning Scene in RViz -->
        <arg name="scene_file" default="$(find pkg_task4)/config/rviz/ur5_2.scene"/>
        <node name = "moveit_publish_scene_from_text" pkg= "moveit_ros_planning" type = "moveit_publish_scene_from_text" args= "$(arg scene_file)"/>
        <!-- Run Task-4 ROS Node for ur5_2 -->
        <node name= "node_t4_2" pkg= "pkg_task4" type="node_t4_2.py" output="screen"/>
    </group>

    <!-- Recording Bag File for Submission -->
    <arg name="record" default="false"/>
    <arg name="rec_name" default="t4.bag"/>

    <group if="$(arg record)">
        <node name="rosbag_record_pick" pkg="rosbag" type="record"
       args="record -O $(find pkg_task4)/bag_files/$(arg rec_name) --chunksize=20 /eyrc/vb/ur5_1/vacuum_gripper/logical_camera/ur5_1 /eyrc/vb/ur5_2/vacuum_gripper/logical_camera/ur5_2" output="screen"/>
     </group>

</launch>
